core/controllers/resources.py

```diff
@@ -86,3 +87,25 @@ def get(self, exploration_id, asset_type, encoded_filename):
             self.response.write(raw)
         except:
             raise self.PageNotFoundException
+
+
+class PromoBarHandler(base.BaseHandler):
+    """The handler for checking if Promobar is enabled and fetching
+    promobar message.
+    """
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    REDIRECT_UNFINISHED_SIGNUPS = False
+
+    @acl_decorators.open_access
+    def get(self):
+        try:
+            self.render_json({
+                'promo_bar_enabled': config_domain.PROMO_BAR_ENABLED.value,
+                'promo_bar_message': config_domain.PROMO_BAR_MESSAGE.value
+            })
+        except AttributeError:
```

Comment:
This shadows underlying errors. Why are we doing the try/except?

Also, @vojtechjelinek -- in an earlier comment, you mentioned that:

> Just to check, what if `config_domain.PROMO_BAR_ENABLED` doesn't exist and `get` gets called?

Just so I understand -- what do you mean by "doesn't exist" in this context, and in what way can that happen?

Reply:
Sorry, I got confused now that I look at it `config_domain.PROMO_BAR_ENABLED` will always exist because it is defined in the root of *config_domain.py*

@abhaygupta97 You can remove the try-except stuff.

Reply:
Done.